x-pack/plugins/cases/public/components/create/index.test.tsx

```diff
@@ -92,94 +84,62 @@ describe('CreateCase case', () => {
       data: sampleTags,
       refetch: fetchTags,
     }));
+
+    appMockRenderer = createAppMockRenderer();
   });
 
-  it('it renders', async () => {
-    const wrapper = mount(
-      <TestProviders>
-        <CreateCase {...defaultProps} />
-      </TestProviders>
-    );
-    await act(async () => {
-      expect(wrapper.find(`[data-test-subj="create-case-submit"]`).exists()).toBeTruthy();
-      expect(wrapper.find(`[data-test-subj="create-case-cancel"]`).exists()).toBeTruthy();
-    });
+  it('renders', async () => {
+    const result = appMockRenderer.render(<CreateCase {...defaultProps} />);
+
+    expect(await result.findByTestId('create-case-submit')).toBeInTheDocument();
+    expect(await result.findByTestId('create-case-cancel')).toBeInTheDocument();
   });
 
   it('should open modal on cancel click', async () => {
-    const wrapper = mount(
-      <TestProviders>
-        <CreateCase {...defaultProps} />
-      </TestProviders>
-    );
+    const result = appMockRenderer.render(<CreateCase {...defaultProps} />);
 
-    wrapper.find(`[data-test-subj="create-case-cancel"]`).first().simulate('click');
+    expect(await result.findByTestId('create-case-cancel')).toBeInTheDocument();
+    userEvent.click(await result.findByTestId('create-case-cancel'));
 
-    await waitFor(() => {
-      expect(
-        wrapper.find(`[data-test-subj="cancel-creation-confirmation-modal"]`).exists()
-      ).toBeTruthy();
-    });
+    expect(await result.findByTestId('cancel-creation-confirmation-modal')).toBeInTheDocument();
   });
 
   it('should confirm cancelation on modal confirm click', async () => {
-    const wrapper = mount(
-      <TestProviders>
-        <CreateCase {...defaultProps} />
-      </TestProviders>
-    );
+    const result = appMockRenderer.render(<CreateCase {...defaultProps} />);
 
-    wrapper.find(`[data-test-subj="create-case-cancel"]`).first().simulate('click');
+    expect(await result.findByTestId('create-case-cancel')).toBeInTheDocument();
```

Comment:
**nit:** here and in a few places below we do

```
expect await result.findByTestId('something-...
userEvent.click(await result.findByTestId('something-...
```

we could prob avoid "waiting" and finding twice with

```
const something = await result.findByTestId('something-...

expect(something).toBeInDocument();
userEvent.click(something);
```